cli/src/commands/split.rs

```diff
@@ -62,16 +65,22 @@ pub(crate) struct SplitArgs {
     )]
     revision: RevisionArg,
     /// Split the revision into two parallel revisions instead of a parent and
-    /// child.
+    /// child
     // TODO: Delete `--siblings` alias in jj 0.25+
     #[arg(long, short, alias = "siblings")]
     parallel: bool,
-    /// Put these paths in the first commit
+    /// Files matching any of these filesets are put in the first commit
     #[arg(
         value_hint = clap::ValueHint::AnyPath,
         add = ArgValueCompleter::new(complete::modified_revision_files),
     )]
-    paths: Vec<String>,
+    filesets: Vec<String>,
+    /// Put matching files into the second commit instead of the first
+    ///
+    /// This is equivalent to negating (`~`) the union (`|`) of the provided
+    /// filesets.
+    #[arg(long, conflicts_with_all = ["parallel", "interactive", "tool"])]
+    invert_fileset: bool,
```

Comment:
I'd like to hear more requests for this feature before we add it. And if we later decide to add it, I think it should be done consistently for all (?) commands that accept filesets.

Reply:
moving the invert to a separate PR

Reply:
comment (not actionable): Re inverting, I'm also concerned that it's not general enough, so I'd like to see a strong argument before adding the `--invert-fileset` command-line flag in the future:

- If it's a common issue, I'd also like to see if there's a way to resolve it by modifying the revset language, so that the operation can work in a composable way, and also work in all other commands that accept filesets.
- For `jj split` in particular, it sounds like there might still be some disagreement about whether the results should go into the first or second commit (likewise for change ID and bookmark). Maybe there's still room to analyze this and decide that the default behavior should be inverted (which would hopefully resolve a certain common case of inverting the fileset as well)?